src/action/shop.js

```diff
@@ -0,0 +1,41 @@
+import { SET_NEW_LIST_PRODUCT, SET_NEW_SEARCH_PARAMS } from "../constants";
+import apiFetch from "../utils/apiFetch";
+import { isEmpty } from "../utils/isEmpty";
+import { stringifySearchParams } from "../utils/stringifySearchParams";
+
+export const getListProducts = newParams => {
+  return async (dispatch, getState) => {
+    const { shop } = getState();
+    const { list, params } = shop;
+    if (list[stringifySearchParams({ ...newParams })]) {
```

Comment:
don't use spred operator with destruction only one variables
it's absurdly!!!

stringifySearchParams({ ...newParams }) -> stringifySearchParams(newParams)